File: java-selenium/src/main/java/com/orangehrm/pages/AdminPage.java
package com.orangehrm.pages;

import com.orangehrm.base.BasePage;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;

public class AdminPage extends BasePage {

    @FindBy(xpath = "//h6[text()='Admin']")
    private WebElement adminPageTitle;

    @FindBy(xpath = "//button[normalize-space()='Add']")
    private WebElement addButton;

    @FindBy(xpath = "//input[@placeholder='Type for hints...']")
    private WebElement searchField;

    @FindBy(xpath = "//button[@type='submit']")
    private WebElement searchButton;

    public AdminPage(WebDriver driver) {
        super(driver);
    }

    public boolean isAdminPageDisplayed() {
        return isDisplayed(adminPageTitle);
    }

    public String getAdminPageTitle() {
        return getText(adminPageTitle);
    }

    public void clickAddButton() {
        click(addButton);
    }

    public void searchUser(String username) {
        sendKeys(searchField, username);
        click(searchButton);
    }
}

File: playwright-typescript/src/pages/admin-page.ts
import { Page, Locator } from '@playwright/test';
import { BasePage } from '../base/base-page';

export class AdminPage extends BasePage {
  private readonly adminPageTitle: Locator;
  private readonly addButton: Locator;
  private readonly searchField: Locator;
  private readonly searchButton: Locator;

  constructor(page: Page) {
    super(page);
    this.adminPageTitle = page.locator('h6:has-text("Admin")');
    this.addButton = page.locator('button:has-text("Add")');
    this.searchField = page.getByPlaceholder('Type for hints...');
    this.searchButton = page.locator('button[type="submit"]');
  }

  async isAdminPageDisplayed(): Promise<boolean> {
    return await this.isDisplayed(this.adminPageTitle);
  }

  async getAdminPageTitle(): Promise<string> {
    return await this.getText(this.adminPageTitle);
  }

  async clickAddButton(): Promise<void> {
    await this.click(this.addButton);
  }

  async searchUser(username: string): Promise<void> {
    await this.fill(this.searchField, username);
    await this.click(this.searchButton);
  }
}


